pipeline/src/test_util.rs

```diff
@@ -110,31 +118,50 @@ pub fn test_halo2(file_name: &str, inputs: Vec<Bn254Field>) {
         .compute_proof()
         .unwrap();
 
+    // Also generate a proof with the composite backend
+    Pipeline::default()
+        .from_file(resolve_test_file(file_name))
+        .with_prover_inputs(inputs.clone())
+        .with_backend(powdr_backend::BackendType::Halo2MockComposite, None)
+        .compute_proof()
+        .unwrap();
+
     // `gen_halo2_proof` is rather slow, because it computes two Halo2 proofs.
     // Therefore, we only run it in the nightly tests.
     let is_nightly_test = env::var("IS_NIGHTLY_TEST")
         .map(|v| v == "true")
         .unwrap_or(false);
     if is_nightly_test {
-        gen_halo2_proof(file_name, inputs)
+        gen_halo2_proof(file_name, inputs.clone(), false);
+        gen_halo2_proof(file_name, inputs, true);
     }
 }
 
 #[cfg(not(feature = "halo2"))]
 pub fn test_halo2(_file_name: &str, _inputs: Vec<Bn254Field>) {}
 
 #[cfg(feature = "halo2")]
-pub fn gen_halo2_proof(file_name: &str, inputs: Vec<Bn254Field>) {
+pub fn gen_halo2_proof(file_name: &str, inputs: Vec<Bn254Field>, composite: bool) {
```

Comment:
Not a fan of boolean arguments like that, if it can't be two different functions can you at least introduce an enum for the `{use_}composite` argument?

Reply:
I split the function.